rst_files/mccall_model.rst

```diff
@@ -347,64 +319,36 @@ Our initial guess :math:`v` is the value of accepting at every given wage
 
 .. code:: julia
 
-    function plot_value_function_seq(plt,
-                                     c = 25,
-                                     β = 0.99,
-                                     num_plots = 6)
-
-        plt = deepcopy(plt)
-        v = collect(w_vals ./ (1 - β))
-        v_next = similar(v)
-
-        for i in 1:num_plots
-            plot!(plt, w_vals, v, label = "iterate $i")
-
-            # Update guess
-            for (j, w) in enumerate(w_vals)
-                stop_val = w / (1 - β)
-                cont_val = c + β * sum(v .* p_vals)
-                v_next[j] = max(stop_val, cont_val)
-            end
-            v[:] .= v_next
-        end
-        return plt
+    # parameters and constant objects
+    w_min, w_max = 10, 60
+    w_vals = range(w_min, w_max, length = n + 1)
+    c = 25
+    β = 0.99
+    num_plots = 6
+    E = expectation(dist)
+
+    # functions and matrix to plot
+    stop_val(w) = w / (1 - β)
+    cont_val(v) = c + β * E * v # using E as a linear operator
+    best_val(w, v) = max(stop_val(w), cont_val(v))
+    vs = repeat(w_vals ./ (1 - β), outer = (1, 6)) # data to fill
```

Comment:
I don't find this very clear.  What does the `outer = (1, 6)` do here?  Why are we willing in a bunch of information for `vs[:, col]` for `col > 1` we are about to replace?  If this is just a question of preallocating a matrix, we could make that more clear.

Reply:
The `outer = (1, 6)` is the Julian syntax for `repeat` now (I'm not a fan of it either). 

I'll make that clearer. We could fill it with `0`s, but the first column needs to be pre-allocated with the first guess, and I figured it was easier to just fill the whole thing.

Reply:
For an `Array`, you can use `repeat(w_vals ./ (1 - β), 1, 6)` # Repeat 1 in the first dimension and 6 times on the second.

Reply:
Sorry, I meant to "What does the `outer = (1, 6)` do?" with the tone of how a reader might express confusion.  I understand (i.e. had to lookup :-) the notation.

My main issue was that we fill in a big array but everything is overwritten except the first row.  The use of `repeat` here masks that, and makes you think that the data in the other rows actually matters.  Better to preallocate the matrix and directly fill in the first row to make that clearer.